### PR TITLE
First attempt at color picker UI

### PR DESCRIPTION
**Pull request by guac420**
The rainbow slider is working, but I can't get the saturation square to render correctly. (probably due to my lacking knowledge in opengl) 
UI
I have two problems right now, both are labeled in the UI image. 

1. The first problem is that the saturation square is not rendering at all eventually my end goal is for it to look like this:
SaturationSquare
The code I have rendering the saturation square is lines 33-42 in src\main\java\me\zeroeightsix\kami\gui\kami\theme\kami\RootColorSquareUI

2. The second problem I have is the spacing between other setting sliders in the UI, I have linked this problem with the way I set the height of the RootColorSquareUI component on line 55 in the same file as the previous problem. I tried setting the height this way because I want the RootColorSquareUI to be square. The original code to set the height that doesn't change the spacing of other settings is commented 

My end goal is to have the saturation square, rainbow slider, a button for rainbow hue cycling, and maybe add a transparency slider.

Sorry for the excess commits I'm still learning how to use Github, the commit with all of the code for the ColorUI is in the latest commit.

### CODE REVIEW

**Comment by dominikaaaa** [comment]
> Sorry for the excess commits I'm still learning how to use Github, the commit with all of the code for the ColorUI is in the latest commit.

pulls show the combination of all the commits, don't worry about that.

**Comment by guac420** [comment]
This is an updated look at the UI.

UI
Both the color slider and saturation square are fully functional, but I am still having the problem with spacing between the sliders below the square. I haven't really tried fixing the issue yet but that is my next goal.

Also, I know the saturation square is kinda big I just have it that big for testing purposes.

**Comment by dominikaaaa** [comment]
Oh that looks great so far. I can try take a look at the bug today

**Comment by guac420** [comment]
Just found bug with config storage of alpha value, I will fix it tomorrow.

**Comment by guac420** [comment]
I'm so bad with Github, please excuse my excessive commits. I fixed the storage of the alpha value in the config but I also committed some other files I didn't want to commit by accident. The bug fix is only in the HSBColorHolder class, everything else is accidental but will be fixed in a future commit.

**Comment by dominikaaaa** [comment]
That's fine lol. I'll squash everything before merging so don't worry about how many commits it takes.

**Comment by dominikaaaa** [comment]
It looks a LOT smoother now, but 3 issues

- [ ] Transparency is fucked
- [ ] Alignment is slightly off
- [ ] Square has a white band at the side because it's using the width of a rectangle, not a square

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/RootColorSliderUI.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    @Override
+    public void renderComponent(ColorSlider component, FontRenderer aa) {
+        glColor4f(toF(GuiC.sliderColour.color.getRed()), toF(GuiC.sliderColour.color.getGreen()), toF(GuiC.sliderColour.color.getBlue()), component.getOpacity());
+        glLineWidth(10f);
+        int width = component.getWidth();
+        int height = component.getHeight();
+        HSBColourHolder value = component.getValue();
+        double w = width * (value.getH());
+        float downscale = 2f;
+        glShadeModel(GL_SMOOTH);
```
Comment (Xiaro) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/RootColorSliderUI.java:31]: To draw smooth line, you should also do:
```suggestion
        glEnable(GL_LINE_SMOOTH);
        glHint(GL_LINE_SMOOTH_HINT, GL_NICEST);
```

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/RootColorSliderUI.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        glBegin(GL_LINE_STRIP);
+            glColor3f(255, 0, 0); //Red
```
Comment (Xiaro) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/RootColorSliderUI.java:33]: The range of float rbg color is 0f - 1f (0 - 255 in int rgb color), Same with lines after

**Comment by guac420** [comment]
Updated Look at UI:
[Video](https://youtu.be/JcrfZHs4igg)

ColorUI

**Comment by l1ving** [comment]
Hi, I'm closing this as won't fix, thanks for your contribution. This will be implemented in the new GUI (#1510), I hope you understand.

**Pull request closed without merging**